src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                // Before:
+                // void Caller(int i, int j)
+                // {
+                //     Callee(Foo(), Bar());
+                // }
+                // void Callee(int i, int j)
+                // {
+                //     DoSomething(i, j);
+                // }
+                // After:
+                // void Caller(int i, int j)
+                // {
+                //     int i1 = Foo();
+                //     int j1 = Bar();
+                //     DoSomething(i1, j1)
+                // }
+                // void Callee(int i, int j)
+                // {
+                //     DoSomething(i, j);
+                // }
+                // Here two declaration is generated.
+                // Another case is
+                // void Caller()
+                // {
+                //     Callee(out int i)
+                // }
+                // void Callee(out int j)
+                // {
+                //     DoSomething(out j, out int i);
+                // }
+                // After:
+                // void Caller()
+                // {
+                //     int i = 10;
+                //     DoSomething(out i, out int i2);
+                // }
+                // void Callee(out int j)
+                // {
+                //     DoSomething(out j, out int i);
+                // }
+                var localDeclarationStatementsNeedInsert = GetLocalDeclarationStatementsNeedInsert(
+                    inlineMethodRefactoringProvider,
+                    syntaxGenerator,
+                    methodParametersInfo.OperationsNeedGenerateDeclarations,
+                    parametersWithVariableDeclarationArgumentToName!,
+                    renameTable);
+
+                // Get a table which the key is the symbol needs replacement work. Value is the replacement syntax node
+                var replacementTable = ComputeReplacementTable(
+                    inlineMethodRefactoringProvider,
+                    calleeMethodSymbol,
+                    methodParametersInfo.ParametersWithLiteralArgument,
+                    methodParametersInfo.ParametersWithDefaultValue,
+                    syntaxGenerator,
+                    renameTable);
+
+                // Do the replacement work within the callee's body so that it can be inserted to the caller later.
+                inlineNode = (TExpression)await ReplaceAllSyntaxNodesForSymbolAsync(
+                   document,
+                   inlineNode,
+                   syntaxGenerator,
+                   root,
+                   replacementTable,
+                   cancellationToken).ConfigureAwait(false);
+
+                // Check if there is await expression. It is used later if the caller should be changed to async
+                var containsAwaitExpression = inlineNode
+                    .DescendantNodesAndSelf()
+                    .Any(node => node != null && syntaxFacts.IsAwaitExpression(node));
```

Comment:
this won't work.  condier:

```c#
void Caller()
{
    var v = Callee();
}

Action Callee()
{
    return async() => await Task.Delay(1000);
}
```

you don't want to make 'Caller' async here.

WHen descending, you do not want to dive into lambdas/local-functions.

Reply:
I have re-imple this part by double checking its containing method.